ERCS/erc-7930.md

```diff
@@ -0,0 +1,248 @@
+- Addresses do not include any indication of the chain to which the address applies. This means chain information must be expressed out of band, and introduces the risk of an address being mistakenly used on a chain it is not valid for. This risk is particularly pronounced for addresses that represent smart contracts.
+- The existing address format provides no mechanism for extension, meaning that there is no natural method to add these features to the existing address format.
+
+Interoperable Addresses build on insights from [ERC-7828], [CAIP-10] and [CAIP-50], offering a unified format which combines:
+- Binding chain specificity (via explicit chain identifiers) to the raw address.
+- Compact & canonical binary format for use on cross-chain message passing and intent declaration.
+- Checksums for name collision mitigation & user error prevention.
+
+Furthermore, this proposal adds a few features to make addresses interoperable with infrastructure at the edges of and beyond the ethereum ecosystem, by supporting the representation of addresses of non-evm chains as well, for the benefit of cross-chain liquidity movements and cross-chain message passing.
+
+### Guarantees this standard aims to provide
+- Any future Interoperable Address will be trivially convertible to Interoperable Address v1 (specified below), which, together with the point below, makes them a canonical representation for users who need to treat them opaquely.
+- Two Interoperable Addresses, converted to this canonical version, will be bitwise-equal if and only if they represent the same _target address_.
+- Checksums are short enough to be compared by humans, and allow for easy differentiation of _target addresses_ independently of any extra data the Interoperable Address may contain.
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Concepts
+Interoperable Address
+: A binary payload which unambiguously identifies a target address and allows conversion to a human-readable name.
+
+Interoperable Name
+: A string representation of an interoperable address, meant to be used by humans.
+
+Target address
+: The (address, chain ID) pair a particular Interoperable Address points to.
+
+### Interoperable Address binary format definition
+In binary format addresses have the following encoding:
+
+```
+┌─────────┬───────────┬──────────────────────┬────────────────┬───────────────┬─────────┐
+│ Version │ ChainType │ ChainReferenceLength │ ChainReference │ AddressLength │ Address │
+└─────────┴───────────┴──────────────────────┴────────────────┴───────────────┴─────────┘
+```
+
+Where:
+
+Version
+: A 2-byte version identifier. In the current standard this must always be the big-endian unsigned int 1. Other versions should be defined in future ERCs.
```

Comment:
The phrasing could be more direct and concise:

```suggestion
: A 2-byte version identifier. For version 1, this MUST be `0x0001` (big-endian). Future versions SHOULD be standardized in separate ERCs.
```

This removes "In the current standard" redundancy and makes the future versioning guidance clearer.